sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
+      }
+    }
+    return processSummary;
+  }
+
+  /**
+   * Sanitize the the detail records and transform the
+   * detail records to individual disbursements.
+   * @param eceFileDetailRecords detail records of ece file.
+   * @returns disbursements to be processed.
+   */
+  private sanitizeAndTransformToDisbursements(
+    eceFileDetailRecords: ECEResponseFileDetail[],
+  ): ECEDisbursements {
+    const disbursements = {} as ECEDisbursements;
+    for (const eceDetailRecord of eceFileDetailRecords) {
+      const errorMessage = eceDetailRecord.getInvalidDataMessage();
+      if (errorMessage) {
+        throw new Error(
+          `${errorMessage} at line ${eceDetailRecord.lineNumber}`,
+        );
+      }
+      const disbursement =
+        disbursements[eceDetailRecord.disbursementIdentifier];
+      if (!disbursement) {
+        disbursements[eceDetailRecord.disbursementIdentifier] = {
+          institutionCode: eceDetailRecord.institutionCode,
+          applicationNumber: eceDetailRecord.applicationNumber,
+          [AWARD_DETAILS_KEY]: [
+            {
+              payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+              isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+              enrolmentConfirmationDate:
+                eceDetailRecord.enrolmentConfirmationDate,
+            },
+          ],
+        };
+      } else {
+        const awardDetails =
+          disbursements[eceDetailRecord.disbursementIdentifier][
+            AWARD_DETAILS_KEY
+          ];
+        awardDetails.push({
+          payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+          isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+          enrolmentConfirmationDate: eceDetailRecord.enrolmentConfirmationDate,
+        });
+      }
+    }
+    return disbursements;
+  }
+
+  /**
+   * Validate the disbursement and application and
+   * update the enrolment status.
+   * @param disbursements disbursements to be processed,
+   * @param auditUser user who confirm or decline the enrolment.
+   * @param processSummary process summary.
+   * @returns count of disbursements processed, successfully updated, skipped and failed.
+   */
+  private async validateAndUpdateEnrolmentStatus(
+    disbursements: ECEDisbursements,
+    auditUserId: number,
+    processSummary: ProcessSummaryResult,
+  ): Promise<{
+    totalDisbursements: number;
+    disbursementsSuccessfullyProcessed: number;
+    disbursementsSkipped: number;
+    duplicateDisbursements: number;
+    disbursementsFailedToProcess: number;
+  }> {
+    const disbursementSchedules = Object.entries(disbursements);
+    const totalDisbursements = disbursementSchedules.length;
+    let disbursementsSuccessfullyProcessed = 0;
+    let disbursementsSkipped = 0;
+    let duplicateDisbursements = 0;
+    let disbursementsFailedToProcess = 0;
+    for (const [
+      disbursementScheduleId,
+      disbursementDetails,
```

Comment:
`disbursementDetails` is defined as `any` and it is causing warnings down the line, for instance, in lines 215, 221, 222.
Can we please create a type for it?

Reply:
My mistake while checking it. It is properly typed.